lib/matplotlib/tests/test_mlab.py

```diff
@@ -1491,6 +1492,32 @@ def check_maxfreq(self, spec, fsp, fstims):
             del fstimst[-1]
             spect[maxind-5:maxind+5] = 0
 
+    class catch_specgram_warnings(object):
```

Comment:
That is because it is defined in a class, so it becomes a method of that class rather than a freestanding function.  You need to hoist it out.

But I agree with @efiring that it'd be better to just try not triggering the warnings in the first place.